File: Week 1.c/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long card_number = get_long("Number: ");
    int count = 0;
    int sum = 0;
    int is_alternate = 0;
    long card_number_temp = card_number;
    long type_card;

    while (card_number_temp > 0)
    {
        int digit = card_number_temp % 10;

        if (is_alternate)
        {
            digit *= 2;
            sum += (digit > 9) ? (digit / 10) + (digit % 10) : digit;
        }
        else
        {
            sum += digit;
        }

        card_number_temp /= 10;
        is_alternate = !is_alternate;
        count++;
    }

    if (sum % 10 == 0)
    {
        for (int x = 0; count - 2 > x; x++)
        {
            card_number /= 10;
            type_card = card_number;
        }

        if (((type_card == 34) || (type_card == 37)) && (count == 15))
        {
            printf("AMEX\n");
        }
        else if (((type_card >= 51) && (type_card <= 55)) && (count == 16))
        {
            printf("MASTERCARD\n");
        }
        else if (((type_card >= 40) && (type_card <= 49)) && ((count == 13) || (count == 16)))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
        return 0;
    }
    printf("INVALID\n");
}

File: 6.python/credit.py
def checkCard(number):
    sum_array = 0

    for i in range(len(number) - 2, -1, -2):
        double = int(number[i]) * 2
        if double > 9:
            sum_array += double // 10 + double % 10
        else:
            sum_array += double

    for i in range(len(number) - 1, -1, -2):
        sum_array += int(number[i])

    return sum_array % 10 == 0

def main():

    number = input("Number: ").strip()
    if number.isdigit():
        if checkCard(number):
            if len(number) == 15 and (number[:2] == "34" or number[:2] == "37"):
                print("AMEX")
            elif len(number) == 16 and number[:2] in ["51", "52", "53", "54", "55"]:
                print("MASTERCARD")
            elif (len(number) == 13 or len(number) == 16) and number[0] == "4":
                print("VISA")
            else:
                print("INVALID")
        else:
            print("INVALID")
    else:
        print("INVALID")  # Se o input não for um número


main()